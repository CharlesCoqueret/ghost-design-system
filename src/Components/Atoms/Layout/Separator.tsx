import React, { CSSProperties } from 'react';
import classnames from 'classnames';

import styles from './Separator.module.scss';

export interface ISeparatorProps {
  /** Additional class (optional, default: undefined) */
  className?: string;
  /** Custom style (optional, default: undefined) */
  style?: CSSProperties;
}

const Separator = (props: ISeparatorProps) => {
  const { className, style } = props;
  return <hr className={classnames(styles.footer, className)} style={style} />;
};

Separator.defaultProps = {
  className: undefined,
  style: undefined,
};

export default Separator;
