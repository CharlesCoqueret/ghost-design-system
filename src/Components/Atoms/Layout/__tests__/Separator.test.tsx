import React from 'react';
import { render } from '@testing-library/react';

import Separator from '../Separator';

describe('Separator Component', () => {
  it('renders properly', () => {
    const { container } = render(<Separator />);

    expect(container).toMatchSnapshot();
  });
});
