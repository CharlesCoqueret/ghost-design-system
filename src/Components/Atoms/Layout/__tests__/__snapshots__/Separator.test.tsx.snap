// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator Component renders properly 1`] = `
<div>
  <hr
    class="footer"
  />
</div>
`;
